### ISSUE TITLE
Inconsistent behaviour with namespaced symbols in queries

### ISSUE THREAD

**Issue by RickMoynihan** [yetanalytics/flint#31]
To show the inconsistency first here's a query (using a syntax-quote) that works:

```clojure
(def prefixes {:dcat (URI. "http://www.w3.org/ns/dcat#")
               :rdf (URI. "http://www.w3.org/1999/02/22-rdf-syntax-ns#")
               :rdfs (URI. "http://www.w3.org/2000/01/rdf-schema#")
               :foaf (URI. "http://xmlns.com/foaf/0.1/")})

(f/format-query `{:prefixes ~prefixes
                    :select :*
                    :where [[?s ?p ?o]]})
```

Which outputs: 

```
...PREFIXES...
SELECT *
WHERE {
    ?s ?p ?o .
}
```

Noting also that the syntax-quote "`" namespaces symbols within it, i.e. this: 

```clojure
`{:prefixes ~prefixes
                             :select :*
                             :where [[?s ?p ?o]]}
```

evals to this: 

```clojure
{:where [[my.ns/?s my.ns/?p my.ns/?o]],
 :prefixes
 {,,, ,,,]},
 :select :*}
```

So it looks like flint is by convenience supporting the use of syntax quote in constructing queries and internally using `name` to strip the namespaces from the query `?variables`.  However if you try it with a query including a property path it fails validation:

```clojure
(def catalog (URI. "http://my.catalog"))

(f/format-query `{:prefixes ~prefixes
                  :select :*
                  :where [[~catalog (cat :dcat/record :foaf/primaryTopic) ?ds]]})
```

It seems this is because the call to `cat` expands into `my.ns/cat` which then fails validation.  If you want to support syntax quote for easily splicing in data, then I think you may also want to remove the namespaces from the symbols here (and likely also other places e.g. filter functions).

For anyone else running into this sort of thing, you can work around it by separating the path expression out and quoting it with the standard `quote` `'` instead:

```clojure
(f/format-query
  (let [path '(cat :dcat/record :foaf/primaryTopic)]
    `{:prefixes ~prefixes
      :select :*
      :where [[~catalog ~path ?ds]
              [?ds ?dsp ?dso]]}))
```

Or more easily like so:

```clojure
(f/format-query
    `{:prefixes ~prefixes
      :select :*
      :where [[~catalog ~'(cat :dcat/record :foaf/primaryTopic) ?ds]
              [?ds ?dsp ?dso]]}
  :pretty? true)
```

**Comment by kelvinqian00** [comment]
Interesting observation @RickMoynihan.

> So it looks like flint is by convenience supporting the use of syntax quote in constructing queries and internally using `name` to strip the namespaces from the query `?variables`.

I will correct that support for using `` ` `` for variables by using `name` was completely unintentional; I only did it in order to coerce symbols into strings to make validation work, which was not needed for the multi-specs I used with property paths or expressions. Ultimately I did not really provide support for `` ` `` since I though people would mainly use `` ' ``, which was clearly an oversight on my part.

**Comment by RickMoynihan** [comment]
>  support for using syntax-quote for variables by using `name` was completely unintentional;

Ok, that makes sense, it's unspecified behaviour.

To be honest I don't know how many other "query as data" systems (e.g. datomic/asami/honeysql) do this coercion of namespaced symbol names; so it may not be a good idea, or rather it may be placing ease over simplicity.

If you don't do it though your syntax to avoid this issue means prefixing everything with `~'` which might be noisy, but in the grand scheme of things is fine... e.g.

```clojure
(f/format-query
    `{:prefixes ~prefixes
      :select :*
      :where [[~catalog ~'(cat :dcat/record :foaf/primaryTopic) ~'?ds]
              [~'?ds ~'?dsp ~'?dso]]}
  :pretty? true)
```

rather than relying on the implementation detail where you call `name` on those variables.

**Comment by kelvinqian00** [comment]
@RickMoynihan Namespaced syntax quoting for symbols is now disallowed as of v0.3.0, as implemented in #37
